pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -973,4 +973,13 @@ self: super: {
   amqp-utils = super.amqp-utils.override {
     amqp = dontCheck super.amqp_0_18_1;
   };
+
+  # depends on haskell-gi-overloading-0.0
+  ltk = doJailbreak super.ltk;
```

Comment:
`haskell-gi-overloading-0.0` is a special magic dependency that implies a certain build configuration in which some gi overloading feature is disabled. I'm pretty sure that you don't want to jailbreak that. Instead, you should probably just pass the proper version (which we have in hackage-packages.nix).

Reply:
Well, I didn't do much haskell at all so far, so this all is magic to me :-D

I only came across this when changing an input much further down the chain, which triggered a rebuild of `ltk` and all the things behind it.

Building `ltk` seems to have always failed, due to the package from hackage depending on `haskell-gi-overloading`, which seems to not satisfy `haskell-gi-overloading-0.0`, which is why I jailbreaked it to use `0.1`

Using both `haskell-gi-overloading-0.0` and `haskell-gi-overloading-0.1` fails when building `leksah`, see comment 2 of this PR, so that's why I jailbreaked `leksah` too, to build both with 0.1, which apparently makes `leksah` build and run.

So what would be the right way to fix this? Overriding `libraryHaskellDepends` for package `ltk` to the `0.1` version? I fear this will still lead to similar problems as in comment 2.

Reply:
I committed a proper fix in d2f9dc4a3b79c575618b09f239c0ec3e26a371d7.